File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        answer = []
        i = 0
        while (i<len(nums)):
            j = i+1
            k = len(nums)-1
            while (j<k):
                total = nums[i]+nums[j]+nums[k]
                if total>0:
                    k-=1
                elif total<0:
                    j+=1
                else:
                    answer.append([nums[i],nums[j],nums[k]])
                    j+=1
                    while (j<k and nums[j]==nums[j-1]):
                        j+=1
            i+=1
            while(i<len(nums) and nums[i]==nums[i-1]):
                i+=1
        return answer

File: Data Structures & Algorithms/reverse-integer/submission-1.cpp
class Solution {
public:
    int reverse(int x) {
        int answer = 0;
        bool negative = false;
        if (x<0){
            x = -x;
            negative = true;
        }
        while (x>0){
            if (answer>(~0 ^ (1<<31)-x%10)/10){
                return 0;
            }
            answer = answer*10+x%10;
            x=x/10;
            cout << answer << endl;
        }
            
        if (negative){
            answer = -answer;
        }
        return answer;
    }
};
